controllers/utils.go

```diff
@@ -0,0 +1,135 @@
+package controllers
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+
+	odhdeploymentv1 "github.com/opendatahub-io/opendatahub-operator/api/v1"
+	"io"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/resource"
+	"net/http"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+)
+
+// downloadManifests function performs following tasks:
+// 1. It takes component URI and only downloads folder specified by component.ManifestFolder field
+// 2. It saves the manifests in the odh-manifests/component-name/ folder
+func downloadManifests(component odhdeploymentv1.Component) error {
```

Comment:
Do we actually need to download manifests ourselves ? I think both helm and kustomize can use URI as source, can't they ?